doc/Ontology.md

```diff
@@ -0,0 +1,208 @@
+# ACM Knowledge Graph Ontology
+
+## Overview
+
+For details on Knowledge Graphs you can reference other resources.  Perhaps
+start with the [README.md](../README.md) in this repository to begin 
+learning more.
+
+To get started with Knowledge Graphs in ACM, we want our nodes to consist
+of clearly defined entities.  Where possible we want to always re-use the
+existing entities that have been defined instead of creating new ones.
+Validation is being put in place to help prevent too many entities from
+being created.  Too many entity types will make knowledge graph contributions 
+and maintenance more difficult.
+
+## Entities
+
+The list of entities are validated from the [Glossary](../glossary.txt) file
+and consist of the common resources shown below.
+
+![](./ontology.png)
+
+See the examples in the [input](../input) folder for examples on how this
+ontology is being used.
+
+## Entity Descriptions
+
+Entities are the nodes in the knowledge graph.  The following is a list of the
+main entities that are being used in the samples.  While more entities can
+be added, they must not duplicate the ones that already exist.
+
+Entity | Description | Samples | Challenges 
+---    | ---         | ---     | ---
+Users | Users of the ACM APIs. This would also include groups and different User types, such as usage by other Operators or teams. | Customer, Developer, QE, Support | 
+API | A public facing API for the system. Commonly this is the ACM Custom Resource Definitions.  This can also be extended to other APIs such as OpenShift APIs or Non K8S APIs like REST, Ansible and cloud platforms. | ManagedCluster, Policy, Placement, Ansible Playbook | This can seem to overlap with Custom Resource and Intermediate Resource which are also API based. 
```

Comment:
`This can seem to overlap with Custom Resource and Intermediate Resource which are also API based.`
Should we rephrase this with 
`This can seem to overlap with Intermediate Resource which are also API based.` (since Custom Resource is not a entity type anymore).

And then we need to solve the overlap issue. Can we say something like - 
As an example, ManagedCluster is API when user deals with ACM only. However in the context of Telco ZTP flow, the ManagedCluster is not handled by the user. It is generated automatically by the process. The user APIs in that case is the SiteConfig (CR). We propose in such cases to `still` label the ManagedCluster as API - but it will be created from another `Processor` as opposed to a `User`. This allows us to easily query and figure out these nodes.

Reply:
I changed this wording